### PR TITLE
SNS + Lambda support  & deleting AWS keys for roles

### PR DESCRIPTION
**Pull request by dmateusp**

### CODE REVIEW

File: airflow/contrib/operators/awslambda_operator.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+from airflow.contrib.hooks.aws_hook import AwsHook
+
+class AWSLambdaOperator(BaseOperator):
+    """
+    Interact with AWS Lambda
+
+    :param function_name: AWS Lambda Function Name
+    :type function_name: str
+    :param region_name: AWS Region Name (example: us-west-2)
```
Comment (tmccartan) [inline airflow/contrib/operators/awslambda_operator.py]: not a critic, but im not sure we have anything deployed to us-west? default is us-east-1

File: airflow/contrib/operators/awssns_operator.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    ui_color = '#c3dae0'
+    client = None
+    arn = None
+
+    @apply_defaults
+    def __init__(self, message, topic_name=None, target_arn=None, phone_number=None, subject=None,
+                 message_structure=None, message_attributes=None, aws_conn_id=None,
+                 region_name=None, **kwargs):
+        super(AWSSnsOperator, self).__init__(**kwargs)
+
```
Comment (tmccartan) [inline airflow/contrib/operators/awssns_operator.py]: indentation question, extra line why is it needed?

File: airflow/contrib/operators/awssns_operator.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+                    Name=self.topic_name
+                )
+            if topic_arn:
+                self.log.info(
+                    'Topic arn: %s',
+                    topic_arn['TopicArn']
+                )
+                params['TopicArn'] = topic_arn['TopicArn']
+
+            if self.target_arn:
```
Comment (tmccartan) [inline airflow/contrib/operators/awssns_operator.py]: Do we need these params to be validated?

File: airflow/hbc_deployment/Dockerfile [hunk 4]
```diff
@@ -10,7 +10,6 @@ ARG AIRFLOW_PIP=https://github.com/gilt/incubator-airflow/archive/hbc_test.zip
 COPY scripts/* ${AIRFLOW_HOME}/scripts/
 COPY airflow.cfg ${AIRFLOW_HOME}/airflow.cfg
 COPY crontab ${AIRFLOW_HOME}/crontab
-COPY aws_credentials ${AIRFLOW_HOME}/.aws/credentials
```
Comment (tmccartan) [inline airflow/hbc_deployment/Dockerfile]: this is a nice change, I like

File: airflow/hbc_deployment/docker-compose.yml [hunk 5]
```diff
@@ -7,22 +7,31 @@ services:
...
     image: airflow:hbc_test
     entrypoint: /usr/local/airflow/scripts/entrypoint_master.sh
+    volumes:
+     - ${HOME}/airflow/dags:/usr/local/airflow/dags
     environment:
      - AIRFLOW__CELERY__RESULT_BACKEND=db+postgresql://airflow:airflow@db:5432/airflow
      - AIRFLOW__CORE__REMOTE_BASE_LOG_FOLDER=s3://data-team-airflow-logs/airflow-master-tester
      - AIRFLOW__CORE__SQL_ALCHEMY_CONN=postgresql://airflow:airflow@db:5432/airflow
+     - HOST_HOSTNAME=${HOSTNAME}
+     - AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID}
```
Comment (tmccartan) [inline airflow/hbc_deployment/docker-compose.yml]: Where are these coming from? The role on the machine should determine the access privileges not Keys

File: airflow/hbc_deployment/test_dags/test_dag.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
```
Comment (tmccartan) [inline airflow/hbc_deployment/test_dags/test_dag.py]: Can these test be run?

File: airflow/contrib/operators/awslambda_operator.py (hunk 1 above)
Comment (dmateusp) [reply airflow/contrib/operators/awslambda_operator.py]: yes, that's copy pasta code from main Airflow repo, I'll fix that :)

File: airflow/contrib/operators/awssns_operator.py (hunk 2 above)
Comment (dmateusp) [reply airflow/contrib/operators/awssns_operator.py]: not needed!

File: airflow/hbc_deployment/docker-compose.yml (hunk 5 above)
Comment (dmateusp) [reply airflow/hbc_deployment/docker-compose.yml]: docker-compose is only used to run Airflow locally on my machine when I'm developing, my bash profiles exports my AWS credentials in the environment

File: airflow/hbc_deployment/test_dags/test_dag.py (hunk 6 above)
Comment (dmateusp) [reply airflow/hbc_deployment/test_dags/test_dag.py]: Yes, if you use docker-compose you can test those dags locally (they're not unit tests or software tests, they're just DAGs used when I'm checking that everything is working fine, you can schedule them locally)

File: airflow/contrib/operators/awssns_operator.py (hunk 3 above)
Comment (dmateusp) [reply airflow/contrib/operators/awssns_operator.py]: it's validated by boto3, the call to AWS itself, not sure I should double down on that

**Pull request merged**